### PR TITLE
feat(L2): deposits support

### PR DESCRIPTION
**Pull request by ilitteri**
**Motivation**

For users to interact within the L2 we need a way for them to deposit assets.

**Description**

TLDR: This PR adds the necessary implementation to support L1->L2 user deposits.

- [Update CommonBridge deposit impl](https://github.com/lambdaclass/ethereum_rust/commit/0d23c312bf826ea5e8c5c4e61e28ceaa17f7321e)
- [Adapts L1Watcher](https://github.com/lambdaclass/ethereum_rust/commit/53db9bdeb2ed13ba43f1891b66dcac87ab69c5de)
- [Add modified evm handler functions to support deposits](https://github.com/lambdaclass/ethereum_rust/commit/04d7326496d1ae8b3a2a275034d2c0189e61835c)

**TODOs**
- The deposit transaction should be prepared in the `DepositInitiated` event.
- The transaction gas cost should be part of the value to mint, to ensure that the user gets the deposited amount and not less than that. Another thing that we can do is to subsidize deposit transactions to avoid this problem.

## Try it yourself

1. Initialize the stack in one console
    ```
    ethereum_rust_l2 stack init
    ```
2. Transfer money to the bridge
    ```
    cast send --private-key 0x385c546456b6a603a1cfcaa9ec9494ba4832da08dd6bcf4de9a71e4a01b74924 --value 1000000000000000000 0x90cD151b6e9500F13240dF68673f3B81245D57d4 --rpc-url http://localhost:8545
    ```
    You should quickly see a `Mint transaction added to mempool` log.
3. Wait to see the above log and wait for the next block to be produced and then ask for the account balance in L2
    ```
    curl --request POST \
      --url http://localhost:1729 \
      --header 'Content-Type: application/json' \
      --data '{
      "jsonrpc": "2.0",
      "id": 1,
      "method": "eth_getBalance",
      "params": ["0x3d1e15a1a55578f7c920884a9943b3b35d0d885b", "latest"]
    }'
    ```

### CODE REVIEW

File: cmd/ethereum_rust/ethereum_rust.rs [hunk 1]
```diff
@@ -164,8 +164,8 @@ async fn main() {
         if #[cfg(feature = "l2")] {
             let l2_operator = ethereum_rust_l2::start_operator(store.clone()).into_future();
             tracker.spawn(l2_operator);
-            let l2_prover = ethereum_rust_l2::start_prover().into_future();
-            tracker.spawn(l2_prover);
+            // let l2_prover = ethereum_rust_l2::start_prover().into_future();
+            // tracker.spawn(l2_prover);
```
Comment (ManuelBilbao) [inline cmd/ethereum_rust/ethereum_rust.rs:168]: Is this intentioned?

Comment (ilitteri) [reply cmd/ethereum_rust/ethereum_rust.rs:168]: YEs

**Review by ManuelBilbao — APPROVED** [review]
LGTM
